app/controllers/api/users_controller.rb

```diff
@@ -1,8 +1,9 @@
 class Api::UsersController < Api::BaseController
 
   def create
-    user = User.create!(users_params.except(:role))
+    user = User.new(users_params.except(:role))
     user.add_role(users_params[:role])
+    user.save!
```

Comment:
Proba hacer lo siguiente: 
```suggestion
    new_user = User.create!(users_params.except(:role)) do |user|
      user.add_role(users_params[:role])
    end

```

No estoy seguro si anda o no